pos_picking_delayed/models/pos_order.py

```diff
@@ -0,0 +1,46 @@
+# coding: utf-8
+# Copyright 2018 - Today Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    has_picking_delayed = fields.Boolean(
+        default=True, help="This checkbox is checked if the generation of"
+        " the picking has been delayed. The picking will be created by cron.")
+
+    @api.model
+    def create_from_ui(self, orders):
+        return super(PosOrder, self.with_context(
+            picking_creation_delayed=True)).create_from_ui(orders)
+
+    def create_picking(self):
+        if self.env.context.get('picking_creation_delayed', False):
+            for order in self:
+                if not order.config_id.picking_creation_delayed:
+                    super(PosOrder, order).create_picking()
+                    order.write({'has_picking_delayed': False})
```

Comment:
I think a more precise control has to be set here, for old pickings, it set the fields to true even if a picking already exists for the order. This causes huge computation load and slow down all the process.